lib/DataObject/Import/Resolver/Filename.php

```diff
@@ -56,21 +58,33 @@ public function resolve(\stdClass $config, int $parentId, array $rowData)
             throw new ImportErrorException('not allowed to import into folder ' . $parent->getFullPath());
         }
 
-        if ($overwrite) {
-            $objectKey = $rowData[$this->getIdColumn($config)];
-        } else {
-            $objectKey = $prefix;
-        }
+        $object = null;
 
         $classId = $config->classId;
         $classDefinition = ClassDefinition::getById($classId);
         $className = 'Pimcore\\Model\\DataObject\\' . ucfirst($classDefinition->getName());
 
-        $intendedPath = $parent->getRealFullPath() . '/' . $objectKey;
-        $object = null;
+        if ($overwrite) {
+            $objectKey = $rowData[$this->getIdColumn($config)];
+            $intendedPath = $parent->getRealFullPath() . '/' . $objectKey;
+        } else {
+            $objectKey = $rowData[$this->getIdColumn($config)];
+            $objectKey = Service::getValidKey($objectKey, "object");
+            $intendedPath = $parent->getRealFullPath() . '/' . $objectKey;
+
+            if (AbstractObject::getByPath($intendedPath)) {
+                $objectKey = $prefix;
```

Comment:
prefix is ignored for new objects
![image](https://user-images.githubusercontent.com/5137917/86569913-6e3d2f00-bf6f-11ea-87b3-d467446937f5.png)
and when object exists(without overwrite check) then it creates new numbered objects with prefix.
![image](https://user-images.githubusercontent.com/5137917/86570111-b8261500-bf6f-11ea-893f-f7c7062bdcfa.png)

Reply:
this numbering should happen if overwrite is not checked and the key already exists

Reply:
I expect it to take key not number when filename resolver strategy is used.